src/component/modal/teamAddMemberModal/index.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  useState, useEffect, useCallback, memo, useMemo,
+} from 'react';
+import {
+  Typography, Box, List, TextField, ListItemText, Modal, ListItemButton,
+} from '@mui/material';
+import { useSearchMembers } from 'query/members';
+import { Member } from 'model/member';
+import { useCreateTeamMember } from 'query/teams';
+
+import * as S from './style';
+
+interface SearchMemberModalProps {
+  open: boolean;
+  onClose: () => void;
+  teamId: number;
+  isLeader: boolean;
+}
+
+const SearchModal = memo(({
+  open, onClose, teamId, isLeader,
+}: SearchMemberModalProps) => {
```

Comment:
마이너?) 리액트에서 컴포넌트를 선언할 때는 다음과 같은 컨벤션을 가지는 편이에요
변수랑 컴포넌트를 구분하기 용이하게 하기 위해서입니다.
그런데 기능상 동일하긴 해서 큰 문제는 안될 것 같아요
```suggestion
export default function SearchModal({
  open, onClose, teamId, isLeader,
}: SearchMemberModalProps) {
```

Reply:
+) memo를 사용했다면 얘기가 달라지긴 하네요
memo를 사용한다면 이렇게 사용할 수 있다고 합니당
```suggestion
function SearchModal({
  open, onClose, teamId, isLeader,
}: SearchMemberModalProps) {

~~~

export default memo(SearchModal);
```